### PR TITLE
Monitoring: Add Node & Pod info in TiDB Grafana

### PR DESCRIPTION
**Pull request by qiffang**
### What problem does this PR solve? 

- Add kubernetes Node and Pod metrics in TiDB Cluster Grafana
- Support lighting monitoring and alert

It is related to [PR](https://github.com/pingcap/monitoring/pull/12) in monitoring repo

### Check List 
Code changes

- Has Helm charts change

Side effects

- Breaking backward compatibility

Tests 

 - Manual test 

Code changes

 - Has Helm charts change

Side effects

 - Breaking backward compatibility

### Does this PR introduce a user-facing change?:
 
 ```
Action required: Users should migrate the configs in values.yaml of previous chart releases to the new values.yaml of the new chart. Otherwise, the monitor pods will be failed when upgrading monitor with the new chart.
For example, configs in old values.yaml:
monitor:
   ...
   imagePullPolicy: IfNotPresent

After migration, new values.yaml should be as below:
monitor:
   ...
    imagePullPolicy: Always
    config:
      K8S_PROMETHEUS_URL: http://prometheus-k8s.monitoring.svc:9090

 ```

### CODE REVIEW

File: charts/tidb-cluster/templates/monitor-deployment.yaml [hunk 1]
```diff
@@ -171,7 +184,7 @@ spec:
           - name: monitor-data
             mountPath: /data
           - mountPath: /etc/grafana/provisioning/datasources
-            name: datasource
+            name: grafana-datasources
```
Comment (tennix) [inline charts/tidb-cluster/templates/monitor-deployment.yaml]: Why change the name?

File: charts/tidb-cluster/values.yaml [hunk 2]
```diff
@@ -359,7 +359,9 @@ monitor:
   storage: 10Gi
   initializer:
     image: pingcap/tidb-monitor-initializer:v3.0.1
-    imagePullPolicy: IfNotPresent
+    imagePullPolicy: Always
+    config:
+      NODE_EXPORTER_PROM_URL: http://prometheus-k8s.monitoring.svc:9090
```
Comment (tennix) [inline charts/tidb-cluster/values.yaml]: I think it should be `K8S_PROMETHEUS_URL`.

File: charts/tidb-cluster/templates/monitor-deployment.yaml [hunk 3]
```diff
@@ -56,12 +56,22 @@ spec:
         env:
         - name: GF_PROVISIONING_PATH
           value: /grafana-dashboard-definitions/tidb
+        - name: GF_DATASOURCE_PATH
+          value: /etc/grafana/provisioning/datasources
         - name: TIDB_CLUSTER_NAME
           value: {{ template "cluster.name" . }}
         - name: TIDB_ENABLE_BINLOG
           value: {{ .Values.binlog.pump.create | default false | quote }}
+        - name: TIDB_ENABLE_IMPORTER
```
[the remark this answers is not in the document]
Comment (qiffang) [reply charts/tidb-cluster/templates/monitor-deployment.yaml]: removed

File: charts/tidb-cluster/templates/monitor-deployment.yaml [hunk 4]
```diff
@@ -56,12 +56,22 @@ spec:
...
+          value: {{ .Values.importer.create | default false | quote }}
         - name:  PROM_CONFIG_PATH
           value: /prometheus-rules
+        - name: GF_K8S_PROMETHEUS_URL
```
[the remark this answers is not in the document]
Comment (qiffang) [reply charts/tidb-cluster/templates/monitor-deployment.yaml]: 不需要的，因为在 入口处（monitoring 接收方）已经添加了 这个默认值，调用方不需要再给默认值了

File: charts/tidb-cluster/values.yaml (hunk 2 above)
Comment (qiffang) [reply charts/tidb-cluster/values.yaml]: 是这个，在 monitor deployment 里面接受的就是
value: {{ .Values.monitor.initializer.config.NODE_EXPORTER_PROM_URL }}

Comment (tennix) [reply charts/tidb-cluster/values.yaml]: Yes, I mean this variable should be named as `K8S_PROMETHEUS_URL`, it's not just node-exporter.

File: charts/tidb-cluster/templates/monitor-deployment.yaml (hunk 1 above)
Comment (qiffang) [reply charts/tidb-cluster/templates/monitor-deployment.yaml]: Done

File: charts/tidb-cluster/values.yaml (hunk 2 above)
Comment (qiffang) [reply charts/tidb-cluster/values.yaml]: Renamed it to `K8S_PROMETHEUS_URL `

File: charts/tidb-cluster/values.yaml [hunk 5]
```diff
@@ -359,7 +359,9 @@ monitor:
   storage: 10Gi
   initializer:
     image: pingcap/tidb-monitor-initializer:v3.0.1
-    imagePullPolicy: IfNotPresent
+    imagePullPolicy: Always
```
Comment (weekface) [inline charts/tidb-cluster/values.yaml]: why change the `imagePullPolicy` to `Always`?

Comment (qiffang) [reply charts/tidb-cluster/values.yaml]: The new tidb operator is incompatible with old image and for now the monitoring image changes frequency, so we think it is better to update the policy with `Always`

**Pull request merged**